### PR TITLE
1.22.4 Release Notes

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/istio/release-builder/pull/1924

### CODE REVIEW

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  
+
+- **Fixed** Do not add pod to ipset if we have a partial failure adding to the dataplane.
+  ([Issue #52218](https://github.com/istio/istio/issues/52218))
+
+- **Fixed** an issue causing resources to incorrectly be reported by `istioctl proxy-status` as `STALE`.
+  ([Issue #51612](https://github.com/istio/istio/issues/51612))
+
+- **Fixed** an issue that can trigger a deadlock when `discoverySelectors` (configured in `MeshConfig`) and a namespace,
+which has an `Ingress` object or a `gateway.networking.k8s.io` `Gateway` object, moves from being selected to unselected.
```
Comment (wilsonwu) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:37]:
```suggestion
- **Fixed** an issue that can trigger a deadlock when `discoverySelectors` (configured in `MeshConfig`) and a namespace,
  which has an `Ingress` object or a `gateway.networking.k8s.io` `Gateway` object, moves from being selected to unselected.
```

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+- **Fixed** inconsistent behavior with the `istio_agent_cert_expiry_seconds` metric.
+
+- **Fixed** the istiod chart installation for older Helm versions (v3.6 and v3.7) by ensuring that `.Values.profile` is set to a string.
+  ([Issue #52016](https://github.com/istio/istio/issues/52016))
+
+- **Fixed** an omission in ztunnel helm charts which resulted in some Kubernetes resources being created without labels
+ 
+- **Fixed** Do not add pod to ipset if we have a partial failure adding to the dataplane.
+  ([Issue #52218](https://github.com/istio/istio/issues/52218))
```
Comment (dhawton) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:30]:
```suggestion
- **Fixed** handling of a a failure adding a pod to the dataplane where the pod was still added to ipset.
  ([Issue #52218](https://github.com/istio/istio/issues/52218))
```

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+- **Fixed** matching multiple service VIPs in ServiceEntry.
+  ([Issue #51747](https://github.com/istio/istio/issues/51747))
+
+- **Fixed** inconsistent behavior with the `istio_agent_cert_expiry_seconds` metric.
+
+- **Fixed** the istiod chart installation for older Helm versions (v3.6 and v3.7) by ensuring that `.Values.profile` is set to a string.
+  ([Issue #52016](https://github.com/istio/istio/issues/52016))
+
+- **Fixed** an omission in ztunnel helm charts which resulted in some Kubernetes resources being created without labels
```
Comment (dhawton) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:27]:
```suggestion
- **Fixed** an omission in ztunnel helm charts which resulted in some Kubernetes resources being created without labels.
```

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ 
+- **Fixed** Do not add pod to ipset if we have a partial failure adding to the dataplane.
+  ([Issue #52218](https://github.com/istio/istio/issues/52218))
+
+- **Fixed** an issue causing resources to incorrectly be reported by `istioctl proxy-status` as `STALE`.
+  ([Issue #51612](https://github.com/istio/istio/issues/51612))
+
+- **Fixed** an issue that can trigger a deadlock when `discoverySelectors` (configured in `MeshConfig`) and a namespace,
+which has an `Ingress` object or a `gateway.networking.k8s.io` `Gateway` object, moves from being selected to unselected.
```
Comment (dhawton) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:36]:
```suggestion
- **Fixed** an issue that can trigger a deadlock when `discoverySelectors` (configured in `MeshConfig`) and a namespace,
  which has an `Ingress` object or a Kubernetes `Gateway` object, would move from being selected to unselected.
```

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+- **Fixed** an issue causing stale endpoints when the same IP address was present in multiple `WorkloadEntries`.
+
+- **Removed** writing the experimental field `GatewayClass.status.supportedFeatures`, as it is unstable in the API.
```
Comment (dhawton) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:40]:
```suggestion
- **Removed** writing the experimental field `GatewayClass.status.supportedFeatures`, as it was unstable in the API.
```

**Review by dhawton — CHANGES_REQUESTED** [review]

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+release: 1.22.4
+---
+
+This release note describes what is different between Istio 1.22.3 and 1.22.4.
+
+{{< relnote >}}
+
+## Changes
+
+- **Fixed** VirtualMachine WorkloadEntry locality label missing during autoregistration.
```
Comment (dhawton) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:16]:
```suggestion
- **Fixed** an issue where the `VirtualMachine` `WorkloadEntry` locality label was missing during auto-registration.
```

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  ([Issue #51800](https://github.com/istio/istio/issues/51800))
+
+- **Fixed** matching multiple service VIPs in ServiceEntry.
```
Comment (dhawton) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:19]:
```suggestion
- **Fixed** matching multiple service VIPs in `ServiceEntry`.
```

File: content/en/news/releases/1.22.x/announcing-1.22.4/index.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+This release note describes what is different between Istio 1.22.3 and 1.22.4.
+
+{{< relnote >}}
+
+## Changes
+
+- **Fixed** an issue where the `VirtualMachine` `WorkloadEntry` locality label was missing during auto-registration.
+  ([Issue #51800](https://github.com/istio/istio/issues/51800))
+
+- **Fixed** matching multiple service VIPs in `ServiceEntry`.
```
Comment (craigbox) [inline content/en/news/releases/1.22.x/announcing-1.22.4/index.md:19]:
```suggestion
- **Fixed** a case where listeners were missing for addresses beyond the first in a `ServiceEntry`.
```

Comment (craigbox) [reply content/en/news/releases/1.22.x/announcing-1.22.4/index.md:19]: this is a nit you can easily ignore

**Review by craigbox — APPROVED** [review]

**Review by dhawton — APPROVED** [review]

**Comment by dhawton** [comment]
Good to merge when you're ready. The cherrypick will need approval/hold until publish.

**Pull request merged**